projects/packages/my-jetpack/_inc/state/resolvers.js

```diff
@@ -1,18 +1,24 @@
 /**
  * External dependencies
  */
+import apiFetch from '@wordpress/api-fetch';
 
 /**
  * Internal dependencies
  */
-import actions from './actions';
+import { REST_API_SITE_PURCHASES_ENDPOINT } from './constants';
 
 const myJetpackResolvers = {
-	*getPurchases() {
-		yield actions.setPurchasesIsFetching( true );
-		const result = yield actions.fetchPurchases();
-		yield actions.setPurchasesIsFetching( false );
-		return actions.setPurchases( result );
+	getPurchases: () => async ( { dispatch } ) => {
+		dispatch.setPurchasesIsFetching( true );
+
+		try {
+			dispatch.setPurchasesIsFetching( false );
```

Comment:
This is right? Does it need to be after the fetch, not?

Reply:
I think it's almost the same, but let's swap it.

Reply:
addressed here https://github.com/Automattic/jetpack/pull/22448/commits/2038b8f1969a583ec40769ce1a749ac55afbe09b